### PR TITLE
Mo

### PR DESCRIPTION
**Pull request by rootleet**
I can do it, manually and not that fast

### CODE REVIEW

File: searx/curladapter.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+
+CURL_SHARE = pycurl.CurlShare()
+CURL_SHARE.setopt(pycurl.SH_SHARE, pycurl.LOCK_DATA_DNS)
+MULTI_HANDLER = pycurl.CurlMulti()
+
+
+def __test_callback(*args):
+    print "callback called"
+
```
Comment (return42) [inline searx/curladapter.py:17]: It is Py2

File: searx/curladapter.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        self.source_ips = cycle(source_ips) if source_ips else cycle([None])
+
+    def add(self, url, **kwargs):
+        handle = get_connection(next(self.source_ips))
+        request_container = RequestContainer(url, handle, **kwargs)
+        try:
+            self._curl_multi_handler.add_handle(handle)
+        except:
+            print 'meep'
```
Comment (return42) [inline searx/curladapter.py:105]: same here

File: searx/curladapter.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+if __name__ == '__main__':
+    r = MultiRequest()
+    r.add('http://httpbin.org/delay/0', headers={'User-Agent': 'x'})
+    r.add('http://127.0.0.1:7777/', headers={'User-Agent': 'x'})
+    r.add('http://httpbin.org/delay/0', cookies={'as': 'sa', 'bb': 'cc'})
+    r.add('http://httpbin.org/delay/0', callback=__test_callback, timeout=1.0, headers={'User-Agent': 'x'})
+    for v in r.send_requests():
+        print v.url
+        print v.response.text
```
Comment (return42) [inline searx/curladapter.py:173]: did you test with python 2?

**Comment by unixfox** [comment]
What is this PR about? I don't understand.

**Comment by dalf** [comment]
I close this PR:
* this is the pycurl branch from this repository.
* this branch is outdated (Python 2 code as @return42 said)
* there is another branch with pycurl: https://github.com/searx/searx/pull/1725

@rootleet , feel free to add comment / explain.

**Pull request closed without merging**